src/ripple/shamap/impl/SHAMapSync.cpp

```diff
@@ -525,16 +514,25 @@ SHAMap::addKnownNode (const SHAMapNodeID& node, Blob const& rawNode,
 
             if (newNode && isInconsistentNode(newNode))
             {
-                return SHAMapAddNode::invalid();
+                state_ = SHAMapState::Invalid;
+                return SHAMapAddNode::useful();
```

Comment:
Returning the result of `SHAMapAddNode::useful()` when we've just set the `state_` to `Invalid` reads a bit oddly, at least in English.  I see this is consistent with the behavior inside the previous `if`, so I have reason to believe this is a good change (or now both conditions will be wrong :smile:).  I am wondering, however, if `SHAMapAddNode::useful()` has the right name.